deployment/README.md

```diff
@@ -2,29 +2,115 @@
+
+  ```bash
+  helm upgrade --set service.type=LoadBalancer --set replicaCount=<replicas> rig reactive-interaction-gateway
+  # or
+  kubectl scale deployment/<deployment_name> --replicas <replicas>
+  ```
+
+2. `kubectl get pods` should list new pods automatically connected to RIG cluster
+
+You can also inspect logs of pods `kubectl logs <pod_name>` to see how they automatically re-balance Kafka consumers (if you are using Kafka) and adapt Proxy APIs from other nodes.
 
 ## Configuration
 
-### General configuration
+### Node host
+
+Every node in cluster needs to be discoverable by other nodes. For that Elixir/Erlang uses so called `long name` or `short name`. We are using `long name` which is formed in the following way `app_name@node_host`. `app_name` is in our case set to `rig` and `node_host` is taken from environment variable `NODE_HOST`. This can be either IP or container alias or whatever that is routable in network by other nodes.
+
+In Kubernetes world it can be set like this:
+
+```yaml
+- name: NODE_HOST
+  valueFrom:
+    fieldRef:
+      fieldPath: status.podIP
+```
 
-1. Node host - Every node in cluster needs to be discoverable by other nodes. For that Elixir/Erlang uses so called `long name` or `short name`. We are using `long name` which is formed in the following way `app_name@node_host`. `app_name` is in our case set to `rig`, but `node_host` is taken from environment variable `NODE_HOST`. This can be either IP or container alias or whatever that is routable in network by other nodes.
+### Node cookie
 
-1. Node cookie - Nodes in Erlang cluster use cookies as a form of authorization/authentication between them. Only nodes with the same cookie can communicate together. It should be ideally some generated hash, set it to `NODE_COOKIE` environment variable.
+Nodes in Erlang cluster use cookies as a form of authorization/authentication between them. Only nodes with the same cookie can communicate together. It should be ideally some generated hash, set it to the `NODE_COOKIE` environment variable.
 
 ### DNS discovery
 
 RIG currently supports distributed deployment via DNS discovery. To make it work, you need to set two environment variables:
 
-1. Discovery type - Currently RIG supports only DNS discovery. To use DNS, set `DISCOVERY_TYPE` to `dns`.
+1. Discovery type - Currently, RIG supports only `dns` discovery. To use DNS, set the `DISCOVERY_TYPE` to `dns`.
 
-1. DNS name (address) - Address where peerage will do discovery for Node host addresses. Value is taken from environment variable `DNS_NAME`.
+2. DNS name (address) - Address where peerage will do a discovery for Node host addresses. Value is taken from the environment variable `DNS_NAME`.
 
 DNS discovery is executed every 5 seconds.
 
+In Kubernetes world it can be set like this:
+
+```yaml
+- name: DISCOVERY_TYPE
+  value: dns
+- name: DNS_NAME
+  value: "reactive-interaction-gateway-service-headless.default.svc.cluster.local"
```

Comment:
I would suggest to "hard-code" the environment variables `DISCOVERY_TYPE` and `DNS_NAME` in the deployment and then set the value of `DNS_NAME` to  `{{ include "reactive-interaction-gateway.fullname" . }}-headless.{{ .Release.Namespace }}.svc.cluster.local`. Then we can also remove most of this documentation to not confuse the user.